libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/PreferenceManager.java

```diff
@@ -0,0 +1,55 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+class PreferenceManager {
+
+  private SharedPreferences preferences;
+  private final Object lock = new Object();
+  private static final String MAPBOX_NAV_PREFERENCES = "mapbox.navigation.preferences.";
+
+  PreferenceManager(Context context, String prefKey) {
+    this.preferences = context.getSharedPreferences(MAPBOX_NAV_PREFERENCES + prefKey, Context.MODE_PRIVATE);
+  }
+
+  public <T> void set(String key, T value) {
+    synchronized (lock) {
+      SharedPreferences.Editor editor = preferences.edit();
+      if (value instanceof Boolean) {
+        editor.putBoolean(key, (Boolean) value);
```

Comment:
are we casting to boxed type value here? but why the api takes primitive as an input?